types/parse/index.d.ts

```diff
@@ -828,15 +855,15 @@ namespace Parse {
         new(attributes?: Attributes): User;
 
         allowCustomUserClass(isAllowed: boolean): void;
-        become(sessionToken: string, options?: UseMasterKeyOption): Promise<User>;
-        current<T extends Attributes>(): User<T> | undefined;
-        currentAsync(): Promise<User | null>;
-        signUp(username: string, password: string, attrs: any, options?: SignUpOptions): Promise<User>;
-        logIn(username: string, password: string, options?: FullOptions): Promise<User>;
-        logOut(): Promise<User>;
-        requestPasswordReset(email: string, options?: SuccessFailureOptions): Promise<User>;
+        become<T extends User>(sessionToken: string, options?: UseMasterKeyOption): Promise<T>;
```

Comment:
Would you mind explaining why you chose `<T extends User>(...): Promise<T>` over
`<T extends Attributes>(...): Promise<User<T>>`? Not saying that it's wrong, just curious...

Reply:
Yep, I actually initially typed it as `<T extends Attributes>` but I changed it so that users would be able to pass a type that extended User but was not User itself. In other words, if someone made a custom User class that extended Parse.User, then I wanted them to be able to have these methods return their custom class instead of the parent Parse.User class.

Open to suggestions, though, if you have any thoughts on that.

Reply:
Ah, of course, I think that sounds very good 👍

Reply:
Hey this works great! Actually this has been always bugging me since forever. Tnx @rdhelms